check/src/typecheck.rs

```diff
@@ -1441,6 +1438,26 @@ fn primitive_type(op_type: &str) -> ArcType {
     }
 }
 
+/// Returns a span appropriate to mark a type error for `e`. For example the example below returns
+/// the `^` span instead of  `~`
```

Comment:
Maybe it should say something about 'returns the span of the innermost type or value binding body'?